src/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -1019,6 +1019,27 @@ void CodeGen::genSSE2Intrinsic(GenTreeHWIntrinsic* node)
             break;
         }
 
+        case NI_SSE2_SetScalarVector128:
+        {
+            assert(baseType == TYP_DOUBLE);
+            assert(op2 == nullptr);
+
+            instruction ins = Compiler::insOfHWIntrinsic(intrinsicID, node->gtSIMDBaseType);
```

Comment:
nit: It might be useful to extract this up-top, like we do in `genSSEIntrinsic`

Reply:
Will fix

Reply:
@tannergooding Checked the pattern and it seems that in Sse instructions are hard coded:

```C++
        case NI_SSE_SetScalarVector128:
        {
            assert(baseType == TYP_FLOAT);
            assert(op2 == nullptr);

            if (op1Reg == targetReg)
            {
                regNumber tmpReg = node->GetSingleTempReg();

                // Ensure we aren't overwriting targetReg
                assert(tmpReg != targetReg);

                emit->emitIns_R_R(INS_movaps, emitTypeSize(TYP_SIMD16), tmpReg, op1Reg);
                op1Reg = tmpReg;
            }

            emit->emitIns_SIMD_R_R_R(INS_xorps, emitTypeSize(TYP_SIMD16), targetReg, targetReg, targetReg);
            emit->emitIns_SIMD_R_R_R(INS_movss, emitTypeSize(TYP_SIMD16), targetReg, targetReg, op1Reg);
            break;
        }
```

I would prefer to not follow that pattern.

Reply:
Hmmm, seems I was thinking of another method.

I was suggesting that it not be hard-coded, but you move the `instruction ins = Compiler::insOfHWIntrinsic` line to the top of `genSSE2Intrinsic`, so it can be more generally shared.

Reply:
Actually, I moved it inside case switches due to explicit code review feedback from @CarolEidt

Reply:
Hmmm - I don't recall giving that feedback (though I don't doubt that I did). It makes sense to pull it out in this case, as nearly all the cases depend upon it. If a large number had a one-to-one mapping, it might make sense to push it into the cases that require it.

---
In reply to: [172245582](https://github.com/dotnet/coreclr/pull/16736#discussion_r172245582) [](ancestors = 172245582)

Reply:
The comment is here: https://github.com/dotnet/coreclr/pull/16262#discussion_r167043783

@CarolEidt  Pls confirm if I should refactor

Reply:
Thanks for the pointer. At that time, there weren't that many cases, so it seemed better to push it down into the cases that needed it, but now there are many cases with the majority requiring it, so it would make sense to pull it up. However, I don't think it's critical.

Reply:
I am keen to get better implementation. May I do it together with next PRs?

Reply:
> I am keen to get better implementation. May I do it together with next PRs?

That would be fine, though we will want to slow the churn in the code as we move closer to release.